CMakeLists.txt

```diff
@@ -67,13 +72,23 @@ if (WIN32)
     string(REPLACE "/MDd" "/MTd" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
 else()
     if (APPLE)
-        set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-pthread")
+        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
     else ()
-        set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-pthread -Wl,--version-script -Wl,${PROJECT_SOURCE_DIR}/icd_exports.map")
+        set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-Wl,--version-script -Wl,${CMAKE_CURRENT_SOURCE_DIR}/loader/linux/icd_exports.map")
+        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
     endif ()
 endif ()
 
+include_directories (${OPENCL_ICD_LOADER_HEADERS_DIR})
+add_definitions (-DCL_TARGET_OPENCL_VERSION=220)
+
+target_include_directories (OpenCL PRIVATE loader)
 target_link_libraries (OpenCL ${CMAKE_DL_LIBS})
 
 enable_testing()
```

Comment:
Could you also `include(CTest)` and wrap the tests with `BUILD_TESTING`?

```cmake
include(CTest) # Adds the BUILD_TESTING option (on by default) and `enable_testing()` if testing is enabled
# ...
if(BUILD_TESTING)
    add_subdirectory(test)
endif()
```

Reply:
Fixed - that's a CMake trick I hadn't seen before.  Thanks!